File: solutions/csharp/perfect-numbers/8/PerfectNumbers.cs
using System;
using System.Linq;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if (number <= 0) throw new ArgumentOutOfRangeException();
        var aliquotSum = GetAliquotSum(number);
        return aliquotSum < number ? Classification.Deficient
            : aliquotSum > number ? Classification.Abundant
            : Classification.Perfect;
    }

    private static int GetAliquotSum(int number) => 
        Enumerable
        .Range(1, number / 2)
        .Where(factor => number % factor == 0)
        .Sum(); 
}

File: solutions/clojure/perfect-numbers/1/src/perfect_numbers.clj
(ns perfect-numbers)

(defn factors-of
  [n]
  (filter #(zero? (rem n %)) (range 1 (inc (/ n 2)))))

(defn aliquot-sum
  [num]
  (reduce + (factors-of num)))

(defn classify
  [num]
  (cond
    (< num 0) (throw (IllegalArgumentException. "Number shouldn't be negative."))
    (< (aliquot-sum num) num) :deficient
    (> (aliquot-sum num) num) :abundant
    (= (aliquot-sum num) num) :perfect))
